src/sentry/models/identity.py

```diff
@@ -36,11 +36,43 @@ class IdentityProvider(Model):
     """
 
     __include_in_export__ = False
+    organization = FlexibleForeignKey("sentry.Organization", blank=True, null=True)
 
+    # To be renamed to "provider" because this is a bad column name
     type = models.CharField(max_length=64)
+    # Replacing "type"
+    provider = models.CharField(max_length=64, null=True)
+
+    # To be renamed to "provider_id", because we're already renaming "type"
+    external_id = models.CharField(max_length=64, null=True)
+    # Replacing "external_id"
+    provider_id = models.CharField(max_length=64, null=True)
     config = EncryptedJsonField()
+
     date_added = models.DateTimeField(default=timezone.now, null=True)
-    external_id = models.CharField(max_length=64, null=True)
+
+    # SSO-specific fields
+    is_sso = models.BooleanField(default=False)
```

Comment:
It's not really safe to add a default like this - The way Django does this is to create the default at the DB level to backfill the column then remove it, and handle the default at the application level. Creating the default at the DB level requires Postgres to lock the table while it rewrites it.

What you can do here is use `SeparateDatabaseAndState` to have Postgres just create the column without the default, but keep the default in Django so that new rows are created as expected. Then since you'll be backfilling anyway, you can fill in the default values as part of that.

Reply:
Updated diff does not have any backfills on this table. Noted for the future.